### ISSUE TITLE
E1010 Invalid GetAtt error message could be better

### ISSUE THREAD

**Issue by alexjurkiewicz** [awslabs/cfn-python-lint#367]
*cfn-lint version: (`cfn-lint --version`)* 0.7.1

Code:
```yaml
  ElasticSearchHostname:
    Value:
      Fn::GetAtt: ElasticSearchDomain.DomainEndpoint
```

(I mixed short and long form)

Current:
```
E1010 Invalid GetAtt E.l.a.s.t.i.c.S.e.a.r.c.h.D.o.m.a.i.n...D.o.m.a.i.n.E.n.d.p.o.i.n.t for resource ElasticSearchHostname
cfn/x.cfn.yaml:342:7
```

Better:
```
E1010 GetAtt expects an array of length 2, not String for resource ElasticSearchHostname
```

(nb, this is also an error in an *output*, not *resource*, but I didn't even notice that until filing the bug report. I guess my eyes skip over everything except the line number)

**Comment by cmmeyer** [comment]
So it looks like we're not properly differentiating against the long form and the short form in YAML.  We'll need to catch that, perhaps as a separate error.

Also, Fn::GetAtt is always retrieving an attribute from a resource (not a property or parameter) -- that's what we're referring to here.  You could argue `resource` is redundant since it's always a resource and we could remove it.
